source/slang/check.cpp

```diff
@@ -1959,6 +1959,11 @@ namespace Slang
                         getSink()->diagnose(attr, Diagnostics::expectedSingleIntArg, attr->name);
                     }
                 }
+                else if (attr.As<PushConstantAttribute>())
```

Comment:
I *think* that the existing logic will check the number of arguments on attributes before calling `validateAttribute()`, so that `validateAttribute()` is only required to check conditions that the generic logic cannot.